### PR TITLE
gh-99894: Ensure the local names don't collide with the test file in traceback suggestion error checking

### PR DESCRIPTION
**Pull request by pablogsal**
* Issue: gh-99894

### CODE REVIEW

**Comment by vstinner** [comment]
This branch doesn't work for me:

```
vstinner@mona$ ./python -m test test_traceback -j20 -F -m test_import_from_error_bad_suggestions_do_not_trigger_for_small_names -v
(...)
0:00:07 load avg: 8.01 [ 36/1] test_traceback failed (1 failure)
test_import_from_error_bad_suggestions_do_not_trigger_for_small_names (test.test_traceback.CPythonSuggestionFormattingTests.test_import_from_error_bad_suggestions_do_not_trigger_for_small_names) ... 
  test_import_from_error_bad_suggestions_do_not_trigger_for_small_names (test.test_traceback.CPythonSuggestionFormattingTests.test_import_from_error_bad_suggestions_do_not_trigger_for_small_names) (name='py') ... FAIL
test_import_from_error_bad_suggestions_do_not_trigger_for_small_names (test.test_traceback.PurePythonSuggestionFormattingTests.test_import_from_error_bad_suggestions_do_not_trigger_for_small_names) ... ok

======================================================================
FAIL: test_import_from_error_bad_suggestions_do_not_trigger_for_small_names (test.test_traceback.CPythonSuggestionFormattingTests.test_import_from_error_bad_suggestions_do_not_trigger_for_small_names) (name='py')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vstinner/python/main/Lib/test/test_traceback.py", line 3172, in test_import_from_error_bad_suggestions_do_not_trigger_for_small_names
    self.assertNotIn("vvv", actual)
AssertionError: 'vvv' unexpectedly found in "ImportError: cannot import name 'py' from 'NkosMVLvvvpdqPze' (/tmp/test_python_xfxl02nj/tmp7l231ock/NkosMVLvvvpdqPze.py)"

$ git show
commit 5612e561a96f4da735e966f527e1e01265564ae1 (HEAD -> pr/99895)
Author: Pablo Galindo <pablogsal@gmail.com>
Date:   Wed Nov 30 11:22:49 2022 +0000

    gh-99894: Ensure the local names don't collide with the test file in traceback suggestion error checking
```

**Comment by vstinner** [comment]
Ah, this PR changes test_getattr_error_bad_suggestions_do_not_trigger_for_small_names(), but not test_import_from_error_bad_suggestions_do_not_trigger_for_small_names().

You should also change test_import_from_error_bad_suggestions_do_not_trigger_for_small_names(), no?

**Comment by pablogsal** [comment]
> Ah, this PR changes test_getattr_error_bad_suggestions_do_not_trigger_for_small_names(), but not test_import_from_error_bad_suggestions_do_not_trigger_for_small_names().
> 
> You should also change test_import_from_error_bad_suggestions_do_not_trigger_for_small_names(), no?

Done!

**Comment by vstinner** [comment]
I ran ``./python -m test test_traceback -j20 -F -m test_import_from_error_bad_suggestions_do_not_trigger_for_small_names -v`` for 9 min 24 sec: 3669 tests OK.

**Comment by vstinner** [comment]
Another test: ``./python -m test test_traceback -j20 -F -v`` ran for 4 min 50 sec: 52 tests OK.

**Comment by vstinner** [comment]
Another test: ``./python -m test test_traceback -j20 -F -m test_getattr_error_bad_suggestions_do_not_trigger_for_small_names -v`` ran for 9 min 28 sec: 5086 tests OK.

File: Lib/test/test_traceback.py [hunk 1]
```diff
@@ -2979,8 +2979,8 @@ class MyClass:
             with self.subTest(name=name):
                 actual = self.get_suggestion(MyClass, name)
                 self.assertNotIn("you mean", actual)
```
Comment (vstinner) [inline Lib/test/test_traceback.py]: Maybe check for ``Did you mean``? Same remark for the other test.

Other test_traceback tests check for this (a little bit longer) pattern.

**Review by vstinner — APPROVED** [review]
LGTM: I tested functionnaly the change, it does fix https://github.com/python/cpython/issues/99894

I have a minor suggestion.

File: Lib/test/test_traceback.py (hunk 1 above)
Comment (vstinner) [inline Lib/test/test_traceback.py]:
```suggestion
                self.assertNotIn("Did you mean", actual)
```

File: Lib/test/test_traceback.py [hunk 2]
```diff
@@ -3169,8 +3169,8 @@ def test_import_from_error_bad_suggestions_do_not_trigger_for_small_names(self):
             with self.subTest(name=name):
                 actual = self.get_import_from_suggestion(code, name)
                 self.assertNotIn("you mean", actual)
```
Comment (vstinner) [inline Lib/test/test_traceback.py]:
```suggestion
                self.assertNotIn("Did you mean", actual)
```

**Comment by vstinner** [comment]
test_threading of "Address sanitizer" failed (env changed). I re-ran this CI job.

**Comment by vstinner** [comment]
Python 3.10 and 3.11 are not affected.

File: Lib/test/test_traceback.py [hunk 3]
```diff
@@ -2978,9 +2978,9 @@ class MyClass:
         for name in ("b", "v", "m", "py"):
             with self.subTest(name=name):
                 actual = self.get_suggestion(MyClass, name)
-                self.assertNotIn("you mean", actual)
-                self.assertNotIn("vvv", actual)
-                self.assertNotIn("mom", actual)
+                self.assertNotIn("Did you mean", actual)
+                self.assertNotIn("'vvv", actual)
```
Comment (kadiwa4) [inline Lib/test/test_traceback.py:2982]: I don't know if this has any effect but there is a `'` missing here

**Pull request merged**